src/DynamoSandbox/Program.cs

```diff
@@ -237,14 +241,36 @@ public static void Main(string[] args)
             try
             {
                 var cmdLineArgs = CommandLineArguments.FromArguments(args);
+                var supportedLocale = new HashSet<string>(new[]
+                        {
+                            "cs-CZ", "de-DE", "en-US", "es-ES", "fr-FR", "it-IT",
+                            "ja-JP", "ko-KR", "pl-PL", "pt-BR", "ru-RU", "zh-CN", "zh-TW"
+                        });
+                string libgLocale;
 
                 if (!string.IsNullOrEmpty(cmdLineArgs.Locale))
                 {
                     // Change the application locale, if a locale information is supplied.
                     Thread.CurrentThread.CurrentUICulture = new CultureInfo(cmdLineArgs.Locale);
                     Thread.CurrentThread.CurrentCulture = new CultureInfo(cmdLineArgs.Locale);
+                    libgLocale = cmdLineArgs.Locale;
+                }
+                else
+                {
+                    // In case no language is specified, libG's locale should be that of the OS.
+                    // There is no need to set Dynamo's locale in this case.
+                    libgLocale = CultureInfo.InstalledUICulture.ToString();
                 }
 
+                // If locale is not supported by Dynamo, default to en-US.
+                if (!supportedLocale.Any(s => s.Equals(libgLocale, StringComparison.InvariantCultureIgnoreCase)))
+                    libgLocale = "en-US";
+
+                // Change the locale that LibG depends on.
+                StringBuilder sb = new StringBuilder("LANGUAGE=");
+                sb.Append(libgLocale.Replace("-", "_"));
+                _putenv(sb.ToString());
+
```

Comment:
Wait, wouldn't it be better if we do this?

```csharp
if (supported ...)
{
    var sb = ...;
    _putenv(sb...);
}
```

Which means we only set locale if we manage to match those?